lighthouse-cli/run.ts

```diff
@@ -141,23 +142,28 @@ export async function runLighthouse(
     url: string, flags: Flags, config: Object|null): Promise<{}|void> {
   let launchedChrome: LaunchedChrome|undefined;
 
+  const shouldGather = flags.gatherMode || (flags.gatherMode == flags.auditMode);
+  const shouldSaveResults = flags.auditMode || (flags.gatherMode == flags.auditMode);
+
   try {
-    launchedChrome = await getDebuggableChrome(flags);
-    flags.port = launchedChrome.port;
+    if (shouldGather) {
+      launchedChrome = await getDebuggableChrome(flags);
+      flags.port = launchedChrome.port;
+    }
     const results = await lighthouse(url, flags, config);
 
     const artifacts = results.artifacts;
     delete results.artifacts;
-
-    await saveResults(results, artifacts!, flags);
-    await launchedChrome.kill();
+    if (shouldSaveResults){
```

Comment:
can this be combined with the logic of whether or not to save already in `saveResults`?

Reply:
done